### PR TITLE
[v14] Simplify `IsBoringCrypto`

### PR DESCRIPTION
**Pull request by espadolini**
Backport #47478 to branch/v14

### CODE REVIEW

**Comment by espadolini** [comment]
I'm going to wait until the first v17 alpha release before merging these backports.

**Review by codingllama — APPROVED** [review]

**Review by tigrato — APPROVED** [review]

**Pull request merged**
